src/js/plugins/search/view.js

```diff
@@ -525,7 +525,8 @@ class SearchView {
       .on(`${namespace}SearchCallbackError`, this.handleSearchCallbackError.bind(this))
       .on(`${namespace}SearchCallbackBookNotIndexed`, this.handleSearchCallbackBookNotIndexed.bind(this))
       .on(`${namespace}SearchCallbackEmpty`, this.handleSearchCallbackEmpty.bind(this))
-      .on(`${namespace}pageChanged`, this.updateSearchNavigation.bind(this));
+      .on(`${namespace}pageChanged`, this.updateSearchNavigation.bind(this))
+      .on(`${namespace}SearchResultsCleared`, this.clearSearchFieldAndResults.bind(this));
```

Comment:
For BookReader's event system, we are using a 1 way directional flow.  Only BookReader emits events, and its consumer will use BR's public functions to update bookreader's state.

This is actually a "mirror" to some of my fixes in my PR #556.

let's sync on Monday to make sure we are on the same page around these quick fixes. thanks!

Reply:
Yeah, I checked your PR that time but just waiting for you to merge so that I can use that event :) and them my whole week off :wink: